code/modules/reagents/reagents/medical.dm

```diff
@@ -16,6 +16,25 @@
 	scannable = TRUE
 	trait_flags = TACHYCARDIC
 
+/datum/reagent/medicine/inaprovaline/on_mob_add(mob/living/L, metabolism)
+	var/health_ratio = L.health / L.maxHealth
+	var/mob/living/carbon/human/H = L
+	if(TIMER_COOLDOWN_CHECK(L, name))
+		return
+	if(health_ratio < -0.5 && volume > 14) //If you are in crit, and someone injects at least 15u into you at once, you will heal 30% of your physical damage instantly.
```

Comment:
this sould not be 0.5, or even a ratio.

IIRC, human have a crit value somewhere, use it

Reply:
I checked and I think the crit value is literally just -0.5 *max health.

Reply:
done